x/concentrated-liquidity/position.go

```diff
@@ -561,21 +566,46 @@ func (k Keeper) GetPositionIdToLock(ctx sdk.Context, positionId uint64) (uint64,
 	return sdk.BigEndianToUint64(value), nil
 }
 
-// SetPositionIdToLock sets the positionId to lock mapping in state.
+// GetPositionIdFromLockId returns the lockId to position mapping in state.
+func (k Keeper) GetPositionIdFromLockId(ctx sdk.Context, underlyingLockId uint64) (uint64, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the lock ID to position ID mapping.
+	key := types.KeyLockIdForPositionId(underlyingLockId)
+	value := store.Get(key)
+	if value == nil {
+		return 0, types.LockIdToPositionIdNotFoundError{LockId: underlyingLockId}
+	}
+
+	return sdk.BigEndianToUint64(value), nil
+}
+
+// SetPositionIdToLock sets both the positionId to lock mapping and the lock to positionId mapping in state.
 func (k Keeper) SetPositionIdToLock(ctx sdk.Context, positionId, underlyingLockId uint64) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
+	// Get the position ID to key mapping and set.
 	key := types.KeyPositionIdForLock(positionId)
 	store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+
+	// Get the lock ID to key mapping and set.
+	key = types.KeyLockIdForPositionId(underlyingLockId)
+	store.Set(key, sdk.Uint64ToBigEndian(positionId))
 }
 
-// RemovePositionIdToLock removes the positionId to lock mapping from state.
+// RemovePositionIdToLock removes both the positionId to lock mapping and the lock to positionId mapping in state.
 func (k Keeper) RemovePositionIdToLock(ctx sdk.Context, positionId uint64) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
+	// Get the position ID to lock mapping.
 	key := types.KeyPositionIdForLock(positionId)
+	underlyingLockId := sdk.BigEndianToUint64(store.Get(key))
+
+	// Delete the mapping from state.
+	store.Delete(key)
+
+	// Get the lock ID to key mapping.
+	key = types.KeyLockIdForPositionId(underlyingLockId)
```

Comment:
Please use different identifiers for keys so that it's clearer what exactly is being removed. Also, checking if the new logic is covered by test?

Reply:
Changed keys here https://github.com/osmosis-labs/osmosis/pull/4885/commits/2c6e24f680b0e1f48c2d06a423db51d50bafce17

Added lock to position check here https://github.com/osmosis-labs/osmosis/pull/4885/commits/439b1799d4f0ea8fadde228f26946b86fe354d30